test/reducer_test.dart

```diff
@@ -2003,17 +1965,13 @@ main() {
     }
   ''';
     Design expected_design = Design.from_json(jsonDecode(expected_json));
-    AppState expected_state = app_state_from_design(expected_design).rebuild((b) => b
-      ..undo_redo.undo_stack.add(simple_strand_design)
-      ..ui_state.changed_since_last_save = true);
-    ;
 
     AppState original_state = app_state_from_design(simple_strand_design)
         .rebuild((b) => b..ui_state.storables.side_selected_helix_idxs = SetBuilder<int>([0, 1]));
 
     AppState final_state = app_state_reducer(original_state, HelixRemoveAllSelected());
 
-    expect_app_state_equal(final_state, expected_state);
+    expect_design_equal(final_state.design, expected_design);
```

Comment:
These tests aren't ideal; e.g. this line is testing the entire design when all that changed was to remove helices. Ideally you'd just test the set of helices to ensure it changed as you expect (without also testing strands, for example). But it's not worth nitpicking for now.

Reply:
I'm kinda on the fence about this. On the one hand, I do see what you mean about not wanting to assert checks for the entire design because yes we want our unit tests to test for specific functionality and not worry about asserting checks on other parts of the design. On the other hand, there is some robustness in assuring that not only did a specified change happen, but also checking that it is the **only** change that happened, which is what doing a full comparison of the designs allows for.

Reply:
> I'm kinda on the fence about this. On the one hand, I do see what you mean about not wanting to assert checks for the entire design because yes we want our unit tests to test for specific functionality and not worry about asserting checks on other parts of the design. On the other hand, there is some robustness in assuring that not only did a specified change happen, but also checking that it is the **only** change that happened, which is what doing a full comparison of the designs allows for.

I see your point. Mostly I'm thinking of times when I've had to fix a unit test after implementing some new feature, but the fix was something incidental, and wasn't actually representing that something broke. It was more like, the definition of correct behavior changed, and the more unit tests there are that reference that behavior, the more fixing has to be done.

One example that springs to mind is the fact that SVG coordinates are stored in helices (which is a design flaw that I'd like to get rid of eventually). Sometimes unit tests would break because the SVG coordinates would change, but it was only because the unit tests hard-coded the old way of computing the SVG coordinates.

Reply:
I see what you mean. I did think it was strange for me to hard code the SVG values when I was writing those tests 😅. For this PR, I rewrote a few tests to get rid of comparing the svg values between helices. Going forward it might make sense to add some helper function that checks for equality between helices that ignores the svg position if we find ourselves needing to write unit tests for comparing helices that have non default svg positions.